### PR TITLE
Fix disable ha-progress button

### PR DESCRIPTION
**Pull request by wendevlin**
## Proposed change
- disable @click on ha-progress-button if the disable property is set.
  - Button was still clickable even if it was disabled.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
Can't we fix this with css?

```css
        pointer-events: none;
```

File: src/components/buttons/ha-progress-button.ts [hunk 1]
```diff
@@ -16,6 +16,12 @@ export class HaProgressButton extends LitElement {
 
   @state() private _result?: "success" | "error";
 
+  constructor() {
+    super();
+
+    this.addEventListener("click", this._buttonTapped);
```
Comment (MindFreeze) [inline src/components/buttons/ha-progress-button.ts:22]: isn't this what is causing `_buttonTapped` to be called when disabled? I don't see why we need another listener

Comment (wendevlin) [reply src/components/buttons/ha-progress-button.ts:22]: This is the listener to the host, _buttonTapped was just listening to the mwc-button.
But the @click event was on the host element, so `_buttonTapped` wasn't even called.

Comment (bramkragten) [reply src/components/buttons/ha-progress-button.ts:22]: Set `pointer-events: none;` on `:host`, it is will no longer create click events

Comment (MindFreeze) [reply src/components/buttons/ha-progress-button.ts:22]: probably remove the `@click` on the button then. It should bubble to the host, I think

Comment (bramkragten) [reply src/components/buttons/ha-progress-button.ts:22]: Yeah, that click handler should not be needed, the button is disabled already

Comment (wendevlin) [reply src/components/buttons/ha-progress-button.ts]: The handler was for checking if it is in progress and then don't forward the click

File: src/components/buttons/ha-progress-button.ts [hunk 2]
```diff
@@ -75,6 +75,7 @@ export class HaProgressButton extends LitElement {
         outline: none;
         display: inline-block;
         position: relative;
+        set-pointer-events: none;
```
Comment (bramkragten) [inline src/components/buttons/ha-progress-button.ts:78]: It is `pointer-events`

File: src/components/buttons/ha-progress-button.ts (hunk 1 above)
Comment (bramkragten) [reply src/components/buttons/ha-progress-button.ts]: > The handler was for checking if it is in progress and then don't forward the click

But the button is disabled in that case too, so should not fire events

Comment (wendevlin) [reply src/components/buttons/ha-progress-button.ts]: Yes

**Comment by wendevlin** [comment]
@MindFreeze @bramkragten `mwc-button` is just there for the style. When I set `pointer-events: none`, they are also not forwarded to the button, so no hover,... styling. 

I switched back to the eventlistener in the constructor but removed the click event listener on the `mwc-button`, because it is not needed.

**Comment by bramkragten** [comment]
@wendevlin 

Doesnt this just work:

```diff
diff --git a/src/components/buttons/ha-progress-button.ts b/src/components/buttons/ha-progress-button.ts
index 357eec1cf..fc04a8b6a 100644
--- a/src/components/buttons/ha-progress-button.ts
+++ b/src/components/buttons/ha-progress-button.ts
@@ -22,7 +22,6 @@ export class HaProgressButton extends LitElement {
       <mwc-button
         ?raised=${this.raised}
         .disabled=${this.disabled || this.progress}
-        @click=${this._buttonTapped}
         class=${this._result || ""}
       >
         <slot></slot>
@@ -63,22 +62,18 @@ export class HaProgressButton extends LitElement {
     }, 2000);
   }
 
-  private _buttonTapped(ev: Event): void {
-    if (this.progress) {
-      ev.stopPropagation();
-    }
-  }
-
   static get styles(): CSSResultGroup {
     return css`
       :host {
         outline: none;
         display: inline-block;
         position: relative;
+        pointer-events: none;
       }
 
       mwc-button {
         transition: all 1s;
+        pointer-events: initial;
       }
 
       mwc-button.success {
```

**Review by MindFreeze — APPROVED** [review]

**Pull request merged**